### PR TITLE
Feature: search bar stops and vehicles

### PR DESCRIPTION
**Pull request by hiba9201**
- Added search bar for stops and vehicles
- Added fly to stops and vehicles in sidebars
- Moved data about units to Redux

### CODE REVIEW

File: client/components/Map/SearchBar/SearchBar.module.css [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+    padding: 12px 16px;
+    border-radius: 4px;
+
+    background-color: #fff;
+
+    flex-direction: row;
+    gap: 8px;
+
+    align-self: stretch;
```
Comment (sashachabin) [inline client/components/Map/SearchBar/SearchBar.module.css:29]: It's default value, maybe remove it?

File: client/components/Map/SearchBar/SearchBar.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+    return (
+        <div className={cn(styles.MapSearchBar)} ref={searchBarRef}>
+            <form
+                className={cn(styles.MapSearchBar__form, {
+                    [styles.MapSearchBar__form_withResult]: hasResults,
+                })}
+                onSubmit={onSubmit}
+            >
+                <img src="/icons/search.svg" alt="Иконка лупы" />
```
Comment (sashachabin) [inline client/components/Map/SearchBar/SearchBar.tsx:109]: Let's set empty `alt=""`

File: client/components/Map/SearchBar/StopsResult/StopResult.module.css [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    align-items: center;
+    cursor: pointer;
+
+    background-color: #fff;
+
+    font-family: 'Iset Sans', sans-serif;
+    font-size: 20px;
+    font-style: normal;
+    font-weight: 400;
+    line-height: 105%;
```
Comment (sashachabin) [inline client/components/Map/SearchBar/StopsResult/StopResult.module.css:13]: Let's use `1.05`

File: client/components/Map/SearchBar/StopsResult/StopResult.module.css [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+    padding: 8px;
+    gap: 10px;
+
+    transition: background-color 150ms ease-out;
+
+    & span {
+        font-size: 28px;
+        line-height: 24px;
```
Comment (sashachabin) [inline client/components/Map/SearchBar/StopsResult/StopResult.module.css:22]: `line-height: 1.16`

File: client/components/Map/SearchBar/StopsResult/StopResult.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        },
+        [dispatch, allStops],
+    );
+
+    return (
+        <div
+            className={cn(styles.MapSearchBarStopResult__wrapper)}
+            onClick={() => setSelectedStop(stopId)}
+        >
+            <img src={`/icons/${type === StopType.TrollBus ? 'bus-troll' : type}.svg`} />
```
Comment (sashachabin) [inline client/components/Map/SearchBar/StopsResult/StopResult.tsx:54]: Let's set empty `alt=""`

File: client/components/Map/SearchBar/UnitResult/UnitResult.tsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+            sidebarService.close();
+        },
+        [dispatch],
+    );
+
+    return (
+        <div
+            className={cn(styles.MapSearchBarUnitResult__wrapper)}
+            onClick={() => setSelectedVehicle(props)}
+        >
```
Comment (sashachabin) [inline client/components/Map/SearchBar/UnitResult/UnitResult.tsx:41]: Let's use `<button>`

File: client/components/Map/SearchBar/StopsResult/StopResult.tsx [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+                onClose: () => dispatch(setCurrentStop(null)),
+            });
+        },
+        [dispatch, allStops],
+    );
+
+    return (
+        <div
+            className={cn(styles.MapSearchBarStopResult__wrapper)}
+            onClick={() => setSelectedStop(stopId)}
```
Comment (sashachabin) [inline client/components/Map/SearchBar/StopsResult/StopResult.tsx:52]: Let's use `<button>`

File: client/components/Map/SearchBar/UnitResult/UnitResult.module.css [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    align-items: center;
+    cursor: pointer;
+
+    background-color: #fff;
+
+    font-family: 'Iset Sans', sans-serif;
+    font-size: 20px;
+    font-style: normal;
+    font-weight: 400;
+    line-height: 105%;
```
Comment (sashachabin) [inline client/components/Map/SearchBar/UnitResult/UnitResult.module.css:13]: 1.05

File: client/components/Map/SearchBar/UnitResult/UnitResult.module.css [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+.MapSearchBarUnitResult__text {
+    margin: 0;
+}
+
+.MapSearchBarUnitResult__text_shallow {
+    color: #55647D;
+    font-size: 16px;
+    font-weight: 300;
+    line-height: 110%;
```
Comment (sashachabin) [inline client/components/Map/SearchBar/UnitResult/UnitResult.module.css:38]: 1.1

File: client/components/Map/Vehicles/Route/MapVehiclesRoute.module.css [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+
+.MapVehiclesRouteSize_s {
+    line-height: 30px;
+    min-width: 50px;
+
+    font-size: 24px;
+}
+
+.MapVehiclesRouteSize_m {
+    line-height: 32px;
```
Comment (sashachabin) [inline client/components/Map/Vehicles/Route/MapVehiclesRoute.module.css:17]: 1.14

File: client/components/Map/Vehicles/Route/MapVehiclesRoute.module.css [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+    min-width: 57px;
+
+    font-size: 28px;
+}
+
+.MapVehiclesRouteSize_l {
+    line-height: 42px;
```
Comment (sashachabin) [inline client/components/Map/Vehicles/Route/MapVehiclesRoute.module.css:24]: 1.31

File: client/components/Map/MainContainer/MapMainContainer.module.css [hunk 12]
```diff
@@ -2,3 +2,21 @@
     width: 100%;
     height: 100%;
 }
+
+.MapSidebar {
+    position: relative;
+    max-height: calc(100vh - 100px);
+    padding: 24px 16px;
+    width: 448px;
```
Comment (sashachabin) [inline client/components/Map/MainContainer/MapMainContainer.module.css:10]: Let's use max-width

`max-width: 448px;`

File: client/components/Map/MainContainer/MapMainContainer.module.css [hunk 13]
```diff
@@ -2,3 +2,21 @@
...
+
+    display: grid;
+    gap: 16px;
+
+    grid-template: fit-content(50%) 1fr / 1fr;
+}
+
+@media screen and (max-width: 768px) {
```
Comment (sashachabin) [inline client/components/Map/MainContainer/MapMainContainer.module.css:18]: Let's remove media

File: client/components/Map/SearchBar/SearchBar.tsx [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+
+    useDisablePropagation(searchBarRef);
+
+    const onFocus = useCallback(() => {
+        setPlaceholder('');
+    }, [setPlaceholder]);
+
+    const onBlur = useCallback(() => {
+        setPlaceholder(INPUT_PLACEHOLDER);
+    }, [setPlaceholder]);
```
Comment (sashachabin) [inline client/components/Map/SearchBar/SearchBar.tsx:41]: Let's move placeholder hide methods to CSS selectors:

```
`:focus::placeholder { opacity: 0 }`
```

File: client/components/Map/SearchBar/SearchBar.tsx [hunk 15]
```diff
@@ -0,0 +1,131 @@
...
+            );
+            const busesSearch = uniqBy(
+                units[ClientUnit.Bus].filter(
+                    (bus) =>
+                        bus.lastStation.toLowerCase().includes(searchText) ||
+                        bus.firstStation.toLowerCase().includes(searchText) ||
+                        bus.num.toLowerCase().includes(searchText),
+                ),
+                uniqUnitsIteratee,
+            );
```
Comment (sashachabin) [inline client/components/Map/SearchBar/SearchBar.tsx:82]: Let's create common toll/tram/bus search functions

File: client/components/Map/SearchBar/SearchBar.module.css [hunk 16]
```diff
@@ -0,0 +1,103 @@
...
+    }
+
+    &:-ms-input-placeholder {
+        color: #000;
+    }
+
+    &::-ms-input-placeholder {
+        color: #000;
+    }
+
```
Comment (sashachabin) [inline client/components/Map/SearchBar/SearchBar.module.css:72]: Let's remove all placeholder selectors. Autoprefixier do in automaticaly

File: client/components/Map/Vehicles/Route/MapVehiclesRoute.module.css [hunk 17]
```diff
@@ -0,0 +1,28 @@
+.MapVehiclesRoute {
+    font-weight: 500;
+    text-align: center;
+    border-radius: 4px;
+    color: white;
+    padding: 0 5px;
+}
+
+.MapVehiclesRouteSize_s {
+    line-height: 30px;
```
Comment (sashachabin) [inline client/components/Map/Vehicles/Route/MapVehiclesRoute.module.css:10]: 1.25

File: client/components/Map/SearchBar/UnitResult/UnitResult.module.css [hunk 18]
```diff
@@ -0,0 +1,39 @@
+.MapSearchBarUnitResult__wrapper {
+    display: flex;
+    flex-direction: row;
+    align-items: center;
+    cursor: pointer;
+
+    background-color: #fff;
+
+    font-family: 'Iset Sans', sans-serif;
```
Comment (sashachabin) [inline client/components/Map/SearchBar/UnitResult/UnitResult.module.css:9]: Let's remove unused font-family

File: client/components/Map/SearchBar/SearchBar.module.css [hunk 19]
```diff
@@ -0,0 +1,103 @@
...
+    border-radius: 4px 4px 0px 0px;
+}
+
+.MapSearchBar__input {
+    width: 100%;
+    padding: 0;
+    border: none;
+    outline: none;
+
+    font-family: 'Iset Sans', sans-serif;
```
Comment (sashachabin) [inline client/components/Map/SearchBar/SearchBar.module.css:42]: Let's use `font-family: inherit`

File: client/components/Map/SearchBar/StopsResult/StopResult.module.css [hunk 20]
```diff
@@ -0,0 +1,32 @@
+.MapSearchBarStopResult__wrapper {
+    display: flex;
+    flex-direction: row;
+    align-items: center;
+    cursor: pointer;
+
+    background-color: #fff;
+
+    font-family: 'Iset Sans', sans-serif;
```
Comment (sashachabin) [inline client/components/Map/SearchBar/StopsResult/StopResult.module.css:9]: Let's remove unused font-family for div

File: client/components/Map/SearchBar/SearchBar.module.css [hunk 21]
```diff
@@ -0,0 +1,103 @@
...
+    font-weight: 400;
+    font-size: 18px;
+    line-height: 24px;
+
+    -moz-appearance: textfield;
+    -webkit-appearance: textfield;
+    appearance: textfield;
```
Comment (sashachabin) [inline client/components/Map/SearchBar/SearchBar.module.css:49]: Let's remove it. Or remove vendor prefixies

File: client/components/Map/MainContainer/MapMainContainer.module.css (hunk 12 above)
Comment (hiba9201) [reply client/components/Map/MainContainer/MapMainContainer.module.css:10]: It doesn't work? Width of elements inside the sidebar is the width of `.MapSidebar`

**Pull request merged**